### PR TITLE
New package: civetweb-1.15

### PR DESCRIPTION
**Pull request by bingulo**
#### Testing the changes
- I tested the changes in this PR: **briefly**

#### New package
- This new package conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements): **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_664-LIBC)

### CODE REVIEW

File: srcpkgs/civetweb/template [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Template file for 'civetweb'
+pkgname=civetweb
+version=1.15
+revision=1
+build_style=gnu-makefile
+make_install_target="install install-headers install-lib install-slib"
+make_use_env=yes
+hostmakedepends="openssl-devel"
```
Comment (classabbyamp) [inline srcpkgs/civetweb/template:8]:
```suggestion
```
is only used for headers/linking, not necessary on host

File: srcpkgs/civetweb/template [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+civetweb-devel_package() {
+	depends="civetweb-${version}_${revision}"
+		short_desc+=' - development files'
+		pkg_install() {
+			vmove usr/include
+				vmove "usr/lib/*.a"
+				vmove "usr/lib/*.so"
+				vmove "usr/lib/*.so.1"
```
Comment (classabbyamp) [inline srcpkgs/civetweb/template:32]:
```suggestion
```
this should be in the main package

File: srcpkgs/civetweb/template [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+pre_install() {
+	vmkdir usr/include
+}
+
+post_install() {
+	vlicense LICENSE.md
+}
+
+civetweb-devel_package() {
+	depends="civetweb-${version}_${revision}"
```
Comment (classabbyamp) [inline srcpkgs/civetweb/template:26]:
```suggestion
	depends="${sourcepkg}>=${version}_${revision}"
```

File: srcpkgs/civetweb/template [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+		short_desc+=' - development files'
+		pkg_install() {
+			vmove usr/include
+				vmove "usr/lib/*.a"
+				vmove "usr/lib/*.so"
+				vmove "usr/lib/*.so.1"
+		}
+}
+
+civetweb-doc_package() {
```
Comment (classabbyamp) [inline srcpkgs/civetweb/template:36]: I don't think there's enough documentation here to warrant a subpkg

```
/usr/share/doc/civetweb/CREDITS.md
/usr/share/doc/civetweb/LICENSE.md
/usr/share/doc/civetweb/README.md
/usr/share/doc/civetweb/RELEASE_NOTES.md
/usr/share/doc/civetweb/SECURITY.md
/usr/share/doc/civetweb/civetweb_64x64.png
/usr/share/doc/civetweb/index.html
```

**Review by classabbyamp — COMMENTED** [review]
there's some extra indentation in your subpackage definitions too

File: srcpkgs/civetweb/template (hunk 4 above)
Comment (bingulo) [reply srcpkgs/civetweb/template:34]: I'll remove the doc subpkg, thx.

File: srcpkgs/civetweb/template [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+civetweb-devel_package() {
+	depends="${sourcepkg}>=${version}_${revision}"
+		short_desc+=' - development files'
+		pkg_install() {
+			vmove usr/include
+			vmove "usr/lib/*.a"
+			vmove "usr/lib/*.so"
+		}
```
Comment (classabbyamp) [inline srcpkgs/civetweb/template:31]:
```suggestion
	short_desc+=' - development files'
	pkg_install() {
		vmove usr/include
		vmove "usr/lib/*.a"
		vmove "usr/lib/*.so"
	}
```

**Comment by classabbyamp** [comment]
you'll need to squash this into 1 commit and force-push

**Comment by bingulo** [comment]
Did I need to add libcivetweb.so.1 to common/shlibs?

**Comment by classabbyamp** [comment]
yes, and I think there should be a `libcivetweb` subpackage too

```sh
libcivetweb_package() {
	short_desc+=" - library"
	pkg_install() {
		vmove usr/lib/*.so.*
	}
}
```

**Comment by bingulo** [comment]
> yes, and I think there should be a `libcivetweb` subpackage too
> 
> ```shell
> libcivetweb_package() {
> 	short_desc+=" - library"
> 	pkg_install() {
> 		vmove usr/lib/*.so.*
> 	}
> }
> ```

Should I add this subpackage as dependency for main package? I tested and didn't find any issue without it.

**Comment by classabbyamp** [comment]
no, it should be automatically detected

**Comment by classabbyamp** [comment]
the devel package should be called libcivetweb-devel now

**Comment by classabbyamp** [comment]
looks decent now, can you squash your commits and force-push?

**Comment by bingulo** [comment]
> looks decent now, can you squash your commits and force-push?

I need to squash all commits from "New package" to now into one?

**Comment by classabbyamp** [comment]
yes, it should all be in a single commit called `New package: civetweb-1.15`

**Comment by classabbyamp** [comment]
hm I think something got messed up there

**Comment by classabbyamp** [comment]
there's still a `libcivetweb-devel/var/db/xbps/pkgdb-0.38.plist` in there

**Comment by bingulo** [comment]
> hm I think something got messed up there

My bad, I ran `git add srcpkgs` and added garbage.

I fixed depency of libcivetweb on libcivetweb-devel, where was wrong (was civetweb).

**Comment by bingulo** [comment]
Now everything should be ok.

**Comment by paper42** [comment]
please remove the commit message body

**Comment by bingulo** [comment]
> please remove the commit message body

done

**Comment by bingulo** [comment]
I'll close this PR cause It's a dependency of another package I built, and it'll be referenced there #38537

**Pull request closed without merging**